### PR TITLE
Use Docker containers to build a test cluster

### LINKED ISSUE #16: Set up a full integration test suite
In order to ensure we don't introduce regressions, we should have a test suite running with realistic scenarios. Currently, setting up the Kafka cluster needed to run the tests is not even fully automated. Ideally, the suite would also allow simulating network failures, partition leader changes, etc.

I'm thinking that a Docker based setup would work the best.

### PR DESCRIPTION
**Pull request by dasch**
Closes #16.

### CODE REVIEW

File: spec/test_cluster.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
+require "docker"
+
+Docker.url = ENV.fetch("DOCKER_HOST")
+
+class TestCluster
+  DOCKER_HOSTNAME = URI(ENV.fetch("DOCKER_HOST")).host
```
Comment (codella) [inline spec/test_cluster.rb]: maybe we should also have a default?

File: spec/test_cluster.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        "Image" => KAFKA_IMAGE,
+        "Links" => ["#{@zookeeper.id}:zk"],
+        "ExposedPorts" => {
+          "9092/tcp" => {}
+        },
+        "Volumes" => {
+          "/var/run/docker.sock" => {}
+        },
+        "Env" => [
+          "KAFKA_ADVERTISED_HOST_NAME=192.168.42.45",
```
Comment (codella) [inline spec/test_cluster.rb]: ip set in stone? :D

**Comment by codella** [comment]
dropped a couple of comments, but nice job anyway!!! :+1:

File: spec/test_cluster.rb (hunk 1 above)
Comment (dasch) [inline spec/test_cluster.rb]: I don't think there's a valid default, at least not on OS X :-/ This ENV var has to be set for the CLI Docker client to function anyway.

File: spec/test_cluster.rb (hunk 2 above)
Comment (dasch) [inline spec/test_cluster.rb]: Nice catch, that should be the Docker host.

**Pull request merged**